server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -444,12 +446,19 @@ public void run()
                                   {
                                     if (cachePopulator != null) {
                                       try {
-                                        Futures.allAsList(cacheFutures).get();
+                                        // For some reason, this can get ahead of the evaluation of the sequence that
+                                        // submits cache population. So we set a timeout here so we don't block the
+                                        // executor queue.
+                                        Futures.allAsList(cacheFutures).get(100, TimeUnit.MILLISECONDS);
```

Comment:
can we make this configurable, or use one of the existing cache timeout values? 100ms seems arbitrary, and if set too low may cause us not to cache the more expensive queries, which are the ones that would most benefit from caching.